nuxeo-drive-elasticsearch/src/main/java/org/nuxeo/drive/elasticsearch/ESAuditChangeFinder.java

```diff
@@ -254,53 +255,54 @@ public long getUpperBound() {
      */
     @Override
     public long getUpperBound(Set<String> repositoryNames) {
-        SearchRequest request = new SearchRequest(getESIndexName())
-                .types(ElasticSearchConstants.ENTRY_TYPE)
-                .searchType(SearchType.DFS_QUERY_THEN_FETCH);
-        SearchSourceBuilder source = new SearchSourceBuilder();
+        SearchRequest request = new SearchRequest(getESIndexName()).types(ElasticSearchConstants.ENTRY_TYPE)
+                                                                   .searchType(SearchType.DFS_QUERY_THEN_FETCH);
+        RangeQueryBuilder filterBuilder = QueryBuilders.rangeQuery("logDate");
         long clusteringDelay = getClusteringDelay(repositoryNames);
         if (clusteringDelay > -1) {
             long lastClusteringInvalidationDate = System.currentTimeMillis() - 2 * clusteringDelay;
-            RangeQueryBuilder filterBuilder = QueryBuilders.rangeQuery("logDate")
-                                                           .lt(lastClusteringInvalidationDate);
-            source.query(QueryBuilders.boolQuery().filter(filterBuilder));
-        } else {
-            source.query(QueryBuilders.matchAllQuery());
+            filterBuilder = filterBuilder.lt(lastClusteringInvalidationDate);
         }
+        SearchSourceBuilder source = new SearchSourceBuilder();
         source.sort("id", SortOrder.DESC).size(1);
-        request.source(source);
-        logSearchRequest(request);
-        SearchResponse searchResponse = getClient().search(request);
-        logSearchResponse(searchResponse);
-        List<LogEntry> entries = new ArrayList<>();
-        SearchHits hits = searchResponse.getHits();
-        ObjectMapper mapper = new ObjectMapper();
-        for (SearchHit hit : hits) {
-            try {
-                entries.add(mapper.readValue(hit.getSourceAsString(), LogEntryImpl.class));
-            } catch (IOException e) {
-                log.error("Error while reading Audit Entry from ES", e);
+        // scroll on previous days with a times 2 step up to 30
+        for (int i = 1; i <= 30; i = i * 2) {
```

Comment:
This is going to search up to 16 days in the past max. Did you want 2^30 days? If so, the logic is wrong.